sanic/mixins/signals.py

```diff
@@ -60,10 +60,16 @@ def decorator(handler: SignalHandler):
 
     def add_signal(
         self,
-        handler,
+        handler: Optional[Callable[..., Any]],
         event: str,
         condition: Dict[str, Any] = None,
     ):
+        if not handler:
+
+            async def noop():
```

Comment:
The nested function definition here might re-introduce app pickling issues, but might be fine to leave it.
(Not sure if users use multi-workers on Windows anymore, thats where the app pickling issues were popping up).

Reply:
It would be helpful if our Windows tests were running right now :unamused: